events-api/src/modules/tickets/tickets.service.ts

```diff
@@ -0,0 +1,109 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { DatabaseService } from '../database/database.service';
+import { CreateTicketReqDto } from './dto/requests/create-ticket.dto';
+import { EventsService } from '../events/events.service';
+import { ApiTags } from '@nestjs/swagger';
+import { TicketPaymentReqDto } from './dto/requests/ticket-payment.dto';
+import { PaymentsService } from '../payments/payments.service';
+
+@Injectable()
+@ApiTags('tickets')
+export class TicketsService {
+  constructor(
+    private readonly databaseService: DatabaseService,
+    private readonly eventsService: EventsService,
+    private readonly paymentsService: PaymentsService,
+  ) {}
+
+  async createTicket(ticketUserId: string, dto: CreateTicketReqDto) {
+    await this.eventsService.verifyEventExistence(dto.ticketEventId);
+
+    const createdTicket = await this.databaseService.ticket.create({
+      data: {
+        ticketQuantity: dto.ticketQuantity,
+        ticketPrice: dto.ticketPrice,
+        ticketEventId: dto.ticketEventId,
+        ticketUserId: ticketUserId,
+      },
+    });
+
+    return createdTicket;
+  }
+
+  async getTickets(ticketUserId: string) {
+    const tickets = await this.databaseService.ticket.findMany({
+      where: {
+        ticketUserId: ticketUserId,
+      },
+    });
+
+    return tickets;
+  }
+
+  async getTicketById(ticketUserId: string, ticketId: string) {
+    const foundTicket = await this.databaseService.ticket.findUnique({
+      where: {
+        ticketUserId: ticketUserId,
+        ticketId: ticketId,
+      },
+    });
+
+    if (!foundTicket) {
+      throw new NotFoundException(`Ticket with id ${ticketId} who belongs to user with id ${ticketUserId} not found`);
```

Comment:
I think that wording here is a bit incorrect

Reply:
> I think that wording here is a bit incorrect

fixed